### ISSUE TITLE
Add subresources to CRD definition

### ISSUE THREAD

**Issue by roksolana-d** [GoogleCloudPlatform/spark-on-k8s-operator#363]
I have built spark-operator from sources using Dockerfile. Also, I have defined a custom object definition with status subresource. However, when I try to access this status using `curl http://localhost:8090/apis/sparkoperator.k8s.io/v1alpha1/namespaces/default/sparkapplications/spark-example/status ` where 8090 port is mapped by kubectl proxy, I get this message:

`the server could not find the requested resource`

How can I enable subresources if they are already defined in the SparkApplication yaml file?
I use Kubernetes with Minikube v0.32.0:
`minikube start --kubernetes-version v1.13.0 --memory 8048 --cpus 3 --feature-gates=CustomResourceSubresources=true`

SparkApplications definition:

```
apiVersion: sparkoperator.k8s.io/v1alpha1
kind: SparkApplication
metadata:
  name: spark-example
  namespace: default
spec:
  type: Scala
  image: gcr.io/ynli-k8s/spark:v2.4.0-SNAPSHOT
  mainClass: org.apache.spark.examples.SparkExample
  mainApplicationFile: http://localhost:8089/spark_k8s_airflow.jar
  mode: cluster
  deps: {}
  driver:
    coreLimit: 1000m
    cores: 0.1
    labels:
      version: 2.4.0
    memory: 1024m
    serviceAccount: default
  executor:
    cores: 1
    instances: 1
    labels:
      version: 2.4.0
    memory: 1024m
  imagePullPolicy: Always
```

**Comment by liyinan926** [comment]
The `status` subresource needs to be added to the CRD definition, not the specification of a custom resource object of that CRD. You need to put the `subresources` field into the `SparkApplication` CRD definition in `manifest/spark-operator-crds.yaml`.

**Comment by roksolana-d** [comment]
@liyinan926 I tried that. I have changed it and built from the Dockerfile again. But the error persists

**Comment by liyinan926** [comment]
You will need to run `kubectl apply -f manifest/spark-operator-crds.yaml` to update the CRDs.

**Comment by roksolana-d** [comment]
@liyinan926 Thanks. I completed such steps, but I get the same error:

1) `eval $(minikube docker-env)`
2) `kubectl apply -f manifest/spark-operator-crds.yaml`
3) `docker build -t spark-operator-new .`
4) `kubectl run --rm -i spark-operator --image=spark-operator-new --image-pull-policy=Never`

Could you tell me what am I doing wrong?

**Comment by roksolana-d** [comment]
Thanks for an advice! I have checked it and found out that the definition of the operator is updated to the previous version each time I build a new operator. So I redefined crds with subresources added after the operator was built. Now the status can be reached.

**Comment by liyinan926** [comment]
Great! We cannot yet add the `status` subresource to the CRDs yet because it's not available prior to Kubernetes 1.11 I believe. So for now you need to manually add it.

**Comment by nishantsh77** [comment]
> Great! We cannot yet add the `status` subresource to the CRDs yet because it's not available prior to Kubernetes 1.11 I believe. So for now you need to manually add it.

@liyinan926 I added status subresource to the CRDs and can see it in "kubectl describe customresourcedefinitions sparkapplications.sparkoperator.k8s.io". But in the code it is still null.

Could please advise me what I am doing wrong ?
